modules/xml/test/src/smithy4s/xml/XmlCodecSpec.scala

```diff
@@ -16,75 +16,85 @@
 
 package smithy4s.xml
 
-import smithy4s.xml.internals.XmlDecoderSchemaVisitor
-import smithy4s.xml.internals.XmlCursor
-import weaver._
+import cats.effect.IO
+import cats.syntax.all._
 import fs2._
 import fs2.data.xml._
 import fs2.data.xml.dom._
-import smithy4s.schema.Schema
-import cats.effect.IO
-import cats.syntax.all._
-import smithy4s.schema.Schema._
-import smithy.api.XmlName
 import smithy.api.XmlAttribute
 import smithy.api.XmlFlattened
+import smithy.api.XmlName
 import smithy4s.ByteArray
 import smithy4s.Hints
 import smithy4s.ShapeId
+import smithy4s.schema.Schema
+import smithy4s.schema.Schema._
+import smithy4s.xml.internals.XmlCursor
+import smithy4s.xml.internals.XmlDecoderSchemaVisitor
+import weaver._
+
+import scala.reflect.ClassTag
 
-object XmlDecodingSpec extends SimpleIOSuite {
+object XmlCodecSpec extends SimpleIOSuite {
+
+  implicit class SchemaOps[A](schema: Schema[A]) {
+    def named(name: String) = schema.withId(ShapeId("default", name))
+    def x = named("x")
+    def n(implicit ct: ClassTag[A]) = named(
```

Comment:
Added this to derive a shapeId from the classtag of the type the Schema works against